MapboxNavigation/UIImage.swift

```diff
@@ -6,17 +6,19 @@ extension UIImage {
     static let shieldURLCache = NSCache<NSString, NSURL>()
     static let shieldImageCache = NSCache<NSString, UIImage>()
     
-    static func shieldKey(_ network: String, number: String, height: CGFloat) -> String {
-        return "\(network)-\(number)-\(Int(height))"
+    static let scale = UIScreen.main.scale
+    
+    static func shieldKey(_ url: URL, height: CGFloat) -> String {
+        return "\(url)-\(Int(height))-\(Int(scale))"
     }
     
     static func cachedShield(_ shieldKey: String) -> UIImage? {
         return UIImage.shieldImageCache.object(forKey: shieldKey as NSString)
     }
     
-    static func shieldImage(_ network: String, number: String, height: CGFloat, scale: CGFloat = UIScreen.main.scale, completion: @escaping (UIImage?) -> Void) {
+    static func shieldImage(_ url: URL, height: CGFloat, completion: @escaping (UIImage?) -> Void) {
```

Comment:
These shield methods are better suited for an extension on VisualInstruction or VisualInstructionComponent than an extension on UIImage. A class shouldn’t be extended with a method just because that method returns an instance of the class.

Reply:
Moved in https://github.com/mapbox/mapbox-navigation-ios/pull/767/commits/48347b61fabf2302e00dece89d41a95cdc72809b